cephadm.yml

```diff
@@ -0,0 +1,149 @@
+---
+
+- name: preconfigure linodes
+  import_playbook: pre-config.yml
+
+- hosts: all
+  become: yes
+  tasks:
+  - name: fetch cephadm
+    delegate_to: localhost
+    register: cephadm
+    run_once: true
+    uri:
+      force: yes
+      mode: 0755
+      return_content: yes
+      url: https://github.com/ceph/ceph/raw/octopus/src/cephadm/cephadm
+
+  - name: distribute cephadm
+    copy:
+      dest: /root/cephadm
+      group: root
+      mode: 0755
+      owner: root
+      content: "{{ cephadm.content }}"
+
+  - name: create /etc/ceph directory
+    file:
+      path: /etc/ceph
+      state: directory
+      owner: root
+      group: root
+      mode: 0755
+
+
+- hosts: mon-000
+  become: yes
+  tasks:
+  - name: check ceph installed
+    stat:
+      path: /etc/ceph/ceph.conf
+    register: ceph_installed
+
+  - name: bootstrap octopus
+    shell: /root/cephadm bootstrap --allow-fqdn-hostname --mon-ip {{ monitor_address }}
+    when: not ceph_installed.stat.exists
+
+  - name: config public network
+    shell: /root/cephadm shell -- ceph config set global public_network 192.168.0.0/16
+
+  - name: config cluster_network
+    shell: /root/cephadm shell -- ceph config set global cluster_network 192.168.0.0/16
+
+  - name: log to files
+    shell: /root/cephadm shell -- ceph config set global log_to_file true
+
+  - name: log cluster to file
+    shell: /root/cephadm shell -- ceph config set global mon_cluster_log_to_file true
+
+  - name: disable stderr logging
+    shell: /root/cephadm shell -- ceph config set global log_to_stderr false
+
+  - name: disable stderr logging of cluster log
+    shell: /root/cephadm shell -- ceph config set global mon_cluster_log_to_stderr false
+
+
+- hosts: mons mgrs osds mdss grafana-servers
+  become: yes
+  tasks:
+  - name: fetch ceph.pub
+    shell: cat /etc/ceph/ceph.pub
+    register: ceph_pub
+    delegate_to: mon-000
+    run_once: true
+
+  - name: register ceph.pub with hosts
+    lineinfile:
+      path: /root/.ssh/authorized_keys
+      line: "{{ ceph_pub.stdout }}"
+
+  - name: get hostname
+    shell: hostname --fqdn
+    register: myhostname
+
+  - name: add hostname to cephadm
+    shell: /root/cephadm shell -- ceph orch host add "{{ myhostname.stdout }}" "{{ ansible_ssh_host }}"
+    delegate_to: mon-000
+
+  - name: label host
+    shell: /root/cephadm shell -- ceph orch host label add "{{ myhostname.stdout }}" "{{ ceph_group }}"
+    delegate_to: mon-000
+
+
+- hosts: osds
+  become: yes
+  tasks:
+  - name: get hostname
+    shell: hostname --fqdn
+    register: myhostname
+
+#FIXME this is not idempotent!!
+  - name: add osd
+    shell: /root/cephadm shell -- ceph orch daemon add osd "{{ myhostname.stdout }}:/dev/sdc"
```

Comment:
```bash
cat <<EOF | ceph orch apply -i 
service_type: osd
spec:
  placement:
    hosts:
    - {{ myhostname.stdout }}
  data_devices:
    paths: 
    - /dev/sdc
EOF
```

should be idempotent

Reply:
````
fatal: [osd-010 -> None]: FAILED! => {
    "changed": true,
    "cmd": "tee /dev/stderr <<EOF | ceph orch apply -i -\nservice_type: osd\nspec:\n  placement:\n    hosts:\n    - li566-143.members.linode.com\n  data_devices:\n    paths:\n    - /dev/sdc\nEOF\n",
    "delta": "0:00:01.536072",
    "end": "2020-07-14 20:52:24.575292",
    "rc": 2,
    "start": "2020-07-14 20:52:23.039220"
}

STDERR:

service_type: osd
spec:
  placement:
    hosts:
    - li566-143.members.linode.com
  data_devices:
    paths:
    - /dev/sdc
Error ENOENT: Failed to validate Drive Group: Feature <spec> is not supported


MSG:

non-zero return code
````

?

Reply:
was backported just recently with https://github.com/ceph/ceph/pull/35537 . Maybe try again?